pkg/graphics/primitive2D.go

```diff
@@ -257,11 +262,49 @@ func NewTriangles(
 	gl.BindVertexArray(0)
 
 	p.position = position
-	p.scale = mgl32.Vec2{1,1}
+	p.scale = mgl32.Vec2{1, 1}
 	p.size = size
 	return p
 }
 
+func NewPolylinePrimitive(position mgl32.Vec3, points []mgl32.Vec2, closed bool) (*Primitive2D) {
+	topLeft, bottomRight := utils.GetBoundingBox(points)
+
+	primitive := &Primitive2D{}
+	primitive.position = position
+	primitive.size = bottomRight.Sub(topLeft)
+	primitive.scale = mgl32.Vec2{1, 1}
+	primitive.shaderProgram = NewShaderProgram(vertexShaderPrimitive2D, "", FragmentShaderSolidColor)
+	primitive.invalidateMatrices()
+
+	// Vertices
+	vertices := make([]float32, 0, len(points)*2)
+	for _, p := range points {
+		// The vertices coordinates are relative to the top left and are scaled by size
+		vertices = append(vertices, (p[0]-topLeft[0])/primitive.size.X(), (p[1]-topLeft[1])/primitive.size.Y())
+	}
+	if closed {
+		// Add the first point again to close the loop
+		vertices = append(vertices, vertices[0], vertices[1])
+	}
+
+	primitive.arrayMode = gl.LINE_STRIP
+	primitive.arraySize = int32(len(vertices) / 2)
+
+	// Build the VAO
+	var vbo uint32
+	gl.GenVertexArrays(1, &primitive.vaoId)
+	gl.BindVertexArray(primitive.vaoId)
+	gl.GenBuffers(1, &vbo)
```

Comment:
i noticed we always generate buffers for every primitive, does this memory ever get released when primitive is destroyed?

Reply:
No, we'll need a Release() function, just like the one in ShaderProgram